### PR TITLE
LPS-137567

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	private void _upgradeDDMTemplateVersion() throws Exception {
+		String selectOrphanTemplateVersions =
+			"select templateVersionId FROM DDMTemplateVersion where not " +
+				"exists (select * from DDMTemplate where " +
+					"DDMTemplate.templateId = DDMTemplateVersion.templateId)";
+
+		String deleteOrphanTemplateVersion =
+			"delete from DDMTemplateVersion where templateVersionId = ?";
+
+		try (Statement selectStatement = connection.createStatement();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java:43]: Hi @marianoalvarosaiz, why should we delete the other versions if the `DDMTemplate` still exists? Shouldn't we keep them for version control? Isn't that the purpose of the DDMTemplateVersion table?

Thanks!

Comment (marianoalvarosaiz) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java:43]: Hi @carolmariaabb,
 yes, that's why I checked **not exists** in _DDMTemplate_.
Kind regards.

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java:43]: Got it now, thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			ResultSet resultSet = selectStatement.executeQuery(
+				selectOrphanTemplateVersions);
+			PreparedStatement preparedStatement =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection, deleteOrphanTemplateVersion)) {
+
```
[the remark this answers is not in the document]
Comment (marianoalvarosaiz) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java:49]: Hi @carolmariaabb,
 OK, I'll remove the private method.

Regarding the removal of variables, it's going to be complicated due to SF. The _selectStatement_ if included as literal in the _executeQuery_ method breaks, due to SF, in four lines, so it makes me create an StringBundler variable. And the delete instruction breaks in a couple of lines making it ugly to separate the = and the ?

Kind regards.

[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java:49]: Ok @marianoalvarosaiz, in that case, please put the variables in the same order as [here](https://github.com/liferay/liferay-portal/blob/3cfd52f3288ede51931aff84024884e1af3c4f5b/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_2_4/DDMContentUpgradeProcess.java#L49-L66), with the `resultSet` in another `Try`. In addition, please use the `PreparedStatement` for both cases (Select and Delete) renaming the variables to `selectPreparedStatement` and `deletePreparedStatement`.

Example:

```
try (
	PreparedStatement selectPreparedStatement = 
		connection.prepareStatement(selectOrphanTemplateVersions);
	PreparedStatement deletePreparedStatement =
		AutoBatchPreparedStatementUtil.concurrentAutoBatch(
			connection, deleteOrphanTemplateVersion)) {

	try (ResultSet resultSet = selectPreparedStatement.executeQuery()) {
		...
	}
	...
}

```

Thanks!

**Comment by marianoalvarosaiz** [comment]
See https://github.com/liferay-forms/liferay-portal/pull/833

**Pull request closed without merging**